packages/cysync-core/src/dialogs/AuthenticateX1Vault/context/index.tsx

```diff
@@ -37,22 +42,46 @@ export interface AuthenticateX1VaultDialogProviderProps {
   children: ReactNode;
 }
 
+const emailRegex = /^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/;
+
 export const AuthenticateX1VaultDialogProvider: FC<
   AuthenticateX1VaultDialogProviderProps
 > = ({ children }) => {
   const lang = useAppSelector(selectLanguage);
   const dispatch = useAppDispatch();
   const deviceRequiredDialogsMap: Record<number, number[] | undefined> = {};
+  const [email, setEmail] = useState<string>('');
+  const [error, setError] = useState<string | null>(null);
+
+  const validateEmail = () => {
+    if (email.length > 0 && !emailRegex.test(email)) {
+      setError('Invalid Email');
+      return;
+    }
```

Comment:
We already have a `validateEmail` function in `utils`.

Reply:
using util's `validateEmail`